### DISCUSSION TITLE
How many native rdkit functions are available in current RDKit.js(2022.3.1) ?

### DISCUSSION THREAD

**Discussion by ShangChien** [rdkit/rdkit#5178] [Q&A] [answered]
Except for the MolDrawOptions series display parameters, how many native methods of RDKit are currently available in this js library?
The usage I know from the demo so far:
```
RDKit.get_mol()
RDKit.get_qmol()
mol.get_substruct_match()
mol.get_substruct_matches()
```
It would be better if there was a list of available methods or functions. 
@ptosco Thank you!

**Comment by ShangChien** [comment]
![2022-04-08 100223](https://user-images.githubusercontent.com/48350636/162348770-45903434-0549-40b8-a571-47fc3f5a64fd.png)
Oh! It seems that I can view it in javascript.
It would be very helpful if there were more usage examples of these functions in the demo.

**Comment by MichelML** [comment, marked as answer]
@ShangChien you're right you can investigate this with the browser developer console, you can also check what is available on the mol and qmol objects like so:

main RDKit module (a lot of noise, but notice a few rdkit related class and functions like Mol, SubstructLibrary, get_mol, get_mol_copy, get_qmol, prefer_coordgen, version, etc):
![image](https://user-images.githubusercontent.com/9539763/162469799-00c2b32a-23ed-4bd9-bb36-bc4a3f8bd965.png)

mol object:
![image](https://user-images.githubusercontent.com/9539763/162468322-a018e056-65bf-4c7f-b468-fc1ba09fe760.png)

qmol object:
![image](https://user-images.githubusercontent.com/9539763/162469401-c2c21d36-b2a1-4c1a-9698-e7a7a67c3bb0.png)

Also available is the `SubstructLibrary` (window.RDKit.SubstructLibrary) that you can look into if you want.

-----------------------

Another place to give you an idea of what is available is in the jswrapper.cpp source code of the minimal lib here https://github.com/rdkit/rdkit/blob/master/Code/MinimalLib/jswrapper.cpp 

------------------------

These two things combined with all available demos:

- https://unpkg.com/@rdkit/rdkit/Code/MinimalLib/dist/GettingStartedInJS.html
- https://unpkg.com/@rdkit/rdkit/Code/MinimalLib/dist/demo.html
- https://michelml.github.io/rdkit-js/examples/vanilla-javascript/
- https://michelml.github.io/rdkit-js/examples/react-example/build/  

Is all there is at the moment unless Greg, Paolo or David says otherwise.

------------------------

All this being said, for sure there is room for improvements in terms of documentation to have a centralized api documentation for each class and functions available would be the best. Wasm doesn't seem to make it easy to automate this, but I'm sure we can put in place some processes to maintain a clean API doc for the minimallib.

**Reply by ShangChien** [reply]
Thank you for your time to explain, best wishes!
